static/app/components/tours/components.tsx

```diff
@@ -0,0 +1,262 @@
+  /**
+   * The className of the wrapper element.
+   */
+  className?: string;
+}
+
+/**
+ * Most implementations of TourElementProps will hook in their own context.
+ */
+export type TourElementProps<T extends TourEnumType> = Omit<
+  BaseTourElementProps<T>,
+  'tourContext'
+>;
+
+export function TourElement<T extends TourEnumType>({
+  children,
+  id,
+  title,
+  description,
+  tourContext,
+  position,
+  className,
+}: BaseTourElementProps<T>) {
+  const theme = useTheme();
+  const {currentStepId, dispatch, orderedStepIds, handleStepRegistration} = tourContext;
+  const stepCount = currentStepId ? orderedStepIds.indexOf(id) + 1 : 0;
+  const stepTotal = orderedStepIds.length;
+  const hasPreviousStep = stepCount > 1;
+  const hasNextStep = stepCount < stepTotal;
+  const isOpen = currentStepId === id;
+
+  const {triggerProps, triggerRef, overlayProps} = useOverlay({
+    isOpen,
+    position,
+    disableTrigger: true,
+  });
+  const element = triggerRef.current;
+
+  useEffect(
+    () => handleStepRegistration({id, element}),
+    [id, element, handleStepRegistration]
+  );
+
+  return (
+    <Fragment>
+      <ElementWrapper
+        className={className}
+        ref={triggerProps.ref}
+        aria-expanded={triggerProps['aria-expanded']}
+      >
+        {children}
+      </ElementWrapper>
+      {isOpen ? (
+        <PositionWrapper zIndex={theme.zIndex.tour.overlay} {...overlayProps}>
+          <TourOverlay animated>
+            <TopRow>
+              <div>
+                {stepCount}/{stepTotal}
+              </div>
+              <TourCloseButton
+                onClick={() => dispatch({type: 'END_TOUR'})}
+                icon={<IconClose style={{color: theme.inverted.textColor}} />}
+                aria-label={t('Close')}
+                borderless
+                size="sm"
+              />
+            </TopRow>
+            <TitleRow>{title}</TitleRow>
+            <div>{description}</div>
+            <ActionBar gap={1}>
+              {hasPreviousStep && (
+                <ActionButton size="xs" onClick={() => dispatch({type: 'PREVIOUS_STEP'})}>
+                  {t('Previous')}
+                </ActionButton>
+              )}
+              {hasNextStep ? (
+                <ActionButton size="xs" onClick={() => dispatch({type: 'NEXT_STEP'})}>
+                  {t('Next')}
+                </ActionButton>
+              ) : (
+                <ActionButton size="xs" onClick={() => dispatch({type: 'END_TOUR'})}>
+                  {t('Finish tour')}
+                </ActionButton>
+              )}
+            </ActionBar>
+          </TourOverlay>
+        </PositionWrapper>
+      ) : null}
+    </Fragment>
+  );
+}
+
+const BlurWindow = styled('div')`
+  content: '';
+  position: absolute;
+  inset: 0;
+  z-index: ${p => p.theme.zIndex.tour.blur};
+  user-select: none;
+  pointer-events: none;
+  backdrop-filter: blur(3px);
+`;
+
+const ElementWrapper = styled('div')`
```

Comment:
```suggestion
const TourTriggerWrapper = styled('div')`
```

I can see that this is pretty opinionated wrt styling, do we expect tour entry points to always stay the same, or would it make sense to expose this in a more flexible way?

Reply:
For now I think I'll keep it as is, but I'm sure designs may come to require some flexibility later on. I'll change the component name though, good call.